### PR TITLE
Create delete task by task_id controller

### PR DESCRIPTION
**Pull request by Mohammed-Sbeata**

### CODE REVIEW

**Review by AyaAljalees — APPROVED** [review]

**Review by KhaledToha — APPROVED** [review]
Good job

File: src/routers/task.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import express, { Router } from 'express';
-import { addTask } from '../controllers';
 import { authCheck } from '../middleware';
+import { addTask, deleteTaskById } from '../controllers';
 
 const taskRouter: Router = express.Router();
 taskRouter.post('/tasks', authCheck, addTask);
+taskRouter.delete('/task/delete', authCheck, deleteTaskById)
```
Comment (nada-abuzaid) [inline src/routers/task.ts:7]:
```suggestion
taskRouter.delete('/task', authCheck, deleteTaskById)

```

File: src/controllers/task.ts [hunk 2]
```diff
@@ -38,5 +39,15 @@ const addTask = (req: Request, res: Response, next: NextFunction) => {
     })
     .catch(() => next(new CustomError(500, 'server error')))
 };
+const deleteTaskById = (req: TokenRequest, res: Response, next: NextFunction) => {
+  const { taskId } = req.params;
```
Comment (nada-abuzaid) [inline src/controllers/task.ts:43]: you get the taskId from the param, how you can access the params here? have you tried to print it?

File: src/controllers/task.ts [hunk 3]
```diff
@@ -38,5 +39,15 @@ const addTask = (req: Request, res: Response, next: NextFunction) => {
     })
     .catch(() => next(new CustomError(500, 'server error')))
 };
+const deleteTaskById = (req: TokenRequest, res: Response, next: NextFunction) => {
+  const { taskId } = req.params;
+  deleteTaskByIdQuery(+taskId)
+    .then(() => {
+      res.status(204).json({
```
Comment (nada-abuzaid) [inline src/controllers/task.ts:46]: 204 status means that is no content returned, so we don't have any response here.

A [204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) (No Content) status code if the action has been enacted and no further information is to be supplied.
A [200](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) (OK) status code if the action has been enacted and the response message includes a representation describing the status.

 read more [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE)

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]

**Review by nada-abuzaid — APPROVED** [review]

**Pull request merged**